spec.html

```diff
@@ -29929,10 +29929,11 @@ <h1>String.prototype.endsWith ( _searchString_ [ , _endPosition_ ] )</h1>
           1. If _endPosition_ is *undefined*, let _pos_ be _len_; else let _pos_ be ? ToInteger(_endPosition_).
           1. Let _end_ be min(max(_pos_, 0), _len_).
           1. Let _searchLength_ be the length of _searchStr_.
+          1. If _searchLength_ is 0, return *true*.
           1. Let _start_ be _end_ - _searchLength_.
-          1. If _start_ is less than 0, return *false*.
-          1. If the sequence of code units of _S_ starting at _start_ of length _searchLength_ is the same as the full code unit sequence of _searchStr_, return *true*.
-          1. Otherwise, return *false*.
+          1. If _start_ &lt; 0, return *false*.
+          1. Let _substring_ be the String value equal to the substring of _S_ consisting of the code units at indices _start_ (inclusive) through _end_ (exclusive).
+          1. Return ! SameValueNonNumeric(_substring_, _searchStr_).
```

Comment:
How do people feel about this? It seemed more natural than the duplication, and more parsimonious than defining and using something like "<dfn>string-equals</dfn>".